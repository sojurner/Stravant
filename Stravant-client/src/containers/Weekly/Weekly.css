.data-point {
  cursor: pointer; }

.area-data-details {
  font-family: daybold;
  font-size: 1.5rem;
  padding: 0.2rem;
  color: bisque;
  border: 3px dotted #b0b0f7;
  background-color: black;
  border-radius: 0.5rem; }

text {
  font-size: 1.2rem;
  font-family: daybold; }

.compare-data-text {
  margin: 2.7rem auto; }

text {
  fill: #f8f9f9; }

.label {
  fill: red;
  margin: 1rem; }

.compare-data-text {
  margin: 0rem auto 2.7rem auto; }

.x-axis {
  position: absolute;
  left: 47.5%;
  top: 83%;
  font-family: daybold;
  font-size: 1.5rem;
  color: #5a5959;
  font-weight: 800; }

.y-axis {
  writing-mode: vertical-rl;
  text-orientation: upright;
  position: fixed;
  left: 49%;
  font-size: 2rem;
  font-family: daybold;
  top: 65%;
  margin: -12.5rem 0 0 -12.5rem;
  background-color: #a5a3a3;
  border-radius: 0.8rem; }

.weekly-title {
  position: absolute;
  font-family: daybold;
  left: 49%;
  font-size: 1.7rem;
  top: 27%;
  color: whitesmoke; }

@media screen and (max-width: 650px) {
  .weekly-title {
    top: 21%; } }

@media screen and (max-width: 510px) {
  .compare-data-text {
    margin: 4rem auto 2.7rem auto; }
  svg {
    margin: 0 0 0 -2rem; }
  .y-axis {
    display: none; }
  .x-axis {
    display: none; }
  .weekly-title {
    top: 20%; } }
