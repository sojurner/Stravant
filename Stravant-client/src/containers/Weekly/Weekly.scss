.data-point {
  cursor: pointer;
}

.area-data-details {
  font-family: daybold;
  font-size: 1.5rem;
  padding: 0.2rem;
  color: bisque;
  border: 3px dotted rgb(176, 176, 247);
  background-color: rgb(0, 0, 0);
  border-radius: 0.5rem;
}

text {
  font-size: 1.2rem;
  font-family: daybold;
}

.compare-data-text {
  margin: auto;
}

text {
  fill: rgb(230, 124, 124);
  y: 500px;
}

.label {
  fill: red;
  margin: 1rem;
}

// @media screen and (max-width: 510px) {
//   svg {
//     width: 20rem;
//     height: 20rem;
//   }
// }
