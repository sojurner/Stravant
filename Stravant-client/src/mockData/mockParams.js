export const tokenParams = [
  'https://www.strava.com/oauth/token',
  {
    body:
      '{"client_id":28236,"client_secret":"829f12d9bdb8ce424edf87deff4551c117d3ed36","code":"r5423rffda4c234ctwtwert2ct"}',
    headers: { 'Content-Type': 'application/json' },
    method: 'POST'
  }
];

export const aggregateParams = [
  'https://www.strava.com/api/v3/athletes/28236/stats',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];

export const weeklyParams = [
  'https://www.strava.com/api/v3/athlete/activities?before=1536123319&after=1536036919',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];
