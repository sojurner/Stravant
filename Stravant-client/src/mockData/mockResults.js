export const aggregateResult = { Biked: 0, Ran: 6.472964574269732, Swam: 0 };

export const tokenResult = {
  access_token: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
  athlete: {
    badge_type_id: 0,
    city: null,
    country: null,
    created_at: '2018-09-04T22:53:48Z',
    email: 'pykim0591@gmail.com',
    firstname: 'Paul',
    follower: null,
    friend: null,
    id: 34515396,
    lastname: 'K',
    premium: false,
    profile:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    profile_medium:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    resource_state: 2,
    sex: 'M',
    state: null,
    summit: false,
    updated_at: '2018-09-07T12:20:05Z',
    username: null
  },
  token_type: 'Bearer'
};

export const weeklyResult = {
  Friday: 3784.3,
  Monday: 0,
  Saturday: 0,
  Sunday: 0,
  Thursday: 0,
  Tuesday: 0,
  Wednesday: 0
};
