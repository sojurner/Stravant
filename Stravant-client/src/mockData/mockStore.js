export const currentUser = {
  info: {
    accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
    userId: 34515396,
    firstName: 'Paul',
    gender: 'M',
    profilePic:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg'
  },
  totalStats: {},
  weeklyStats: {}
};

export const pomStatus = {
  pomStatus: true,
  lastPom: '',
  pomHistory: {
    'Tue, Sep 11, 2018 8:39 PM': { second: 19, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 8:40 PM': { second: 3, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 10:22 PM': { second: 3, minute: 0, hour: 0 }
  }
};

export const clubInfo = {
  clubInfo: { clubName: 'Stravant Club', clubId: 473825 },
  clubActivity: {
    mostRecent: {
      name: 'Pamela',
      distance: 4.270478558110628,
      activity: 'Morning Run'
    },
    Pamela: {
      totalDistance: 62.88999999999999,
      totalTime: 635,
      membersRecent: 'Afternoon Run'
    },
    Paul: { totalDistance: 6.46, totalTime: 78, membersRecent: 'Morning Run' },
    Jesse: {
      totalDistance: 277.78000000000003,
      totalTime: 5292,
      membersRecent: 'Morning Run'
    }
  }
};

export const mockStore = {
  ...currentUser,
  ...pomStatus,
  ...clubInfo
};
