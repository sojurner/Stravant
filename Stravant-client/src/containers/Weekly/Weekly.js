import React, { Component } from 'react';
import { AreaChart } from 'react-easy-chart';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './Weekly.css';
import * as apiCalls from '../../helpers/apiCalls/apiCalls';
import * as userActions from '../../actions/userAction';
export class Weekly extends Component {
  constructor() {
    super();
    this.state = {
      xCoordinates: 0,
      yCoordinates: 0,
      dataDisplay: '',
      toggleDisplay: false
    };
  }

  componentDidMount() {
    this.retrieveWeeklyStats();
  }

  retrieveWeeklyStats = async () => {
    const { currentUser, setWeeklyStats } = this.props;
    const accumulativeStats = await apiCalls.getWeeklyStats(
      currentUser.info.accessToken,
      0
    );
    setWeeklyStats(accumulativeStats);
  };

  trackMouseCoordinates = event => {
    this.setState({ xCoordinates: event.screenX, yCoordinates: event.screenY });
  };

  handleClick = event => {
    this.setState({
      xCoordinates: event.pageX,
      yCoordinates: event.pageY,
      dataDisplay: `${event.y} miles`,
      toggleDisplay: !this.state.toggleDisplay
    });
  };

  render() {
    const { yCoordinates, xCoordinates } = this.state;
    const { weeklyStats } = this.props.currentUser;
    const dataToPlot = Object.keys(weeklyStats)
      .map(day => {
        return {
          x: day.slice(0, 3),
          y: Math.round((weeklyStats[day] / 1609) * 100) / 100
        };
      })
      .reverse();
    const styles = {
      position: 'absolute',
      top: yCoordinates - 140,
      left: xCoordinates + 10
    };

    return (
      <div className="compare-data-text" onClick={this.trackMouseCoordinates}>
        {this.props.currentUser.weeklyStats === {} && (
          <img src="https://cdn-images-1.medium.com/max/1200/1*fFC0LdzpExlP3VG93coZDA.gif" />
        )}
        <h3 className="weekly-title">Weekly Progress</h3>
        <AreaChart
          xType={'text'}
          margin={{ top: 10, right: 10, bottom: 60, left: 60 }}
          yDomainRange={[0, 2.4]}
          axes
          grid
          dataPoints
          areaColors={['orange']}
          verticalGrid
          clickHandler={this.handleClick}
          width={375}
          height={450}
          interpolate={'cardinal'}
          data={[dataToPlot]}
        />
        <p className="x-axis"> Day of Week </p>
        <p className="y-axis">Distance</p>

        <p className="area-data-details" style={styles}>
          {this.state.dataDisplay}
        </p>
      </div>
    );
  }
}

const { object, func } = PropTypes;

Weekly.propTypes = {
  currentUser: object,
  setWeeklyStats: func
};

export const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export const mapDispatchToProps = dispatch => ({
  setWeeklyStats: stats => dispatch(userActions.setWeeklyStats(stats))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Weekly);
