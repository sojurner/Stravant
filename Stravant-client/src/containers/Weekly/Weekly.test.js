import React from 'react';
import { Weekly, mapStateToProps, mapDispatchToProps } from './Weekly';
import * as store from '../../mockData/mockStore';

jest.mock('../../helpers/apiCalls/apiCalls');

describe('Weekly', () => {
  let wrapper;
  let mockStore;
  let mockSetWeekly;

  beforeEach(() => {
    mockStore = store.currentUser;
    mockSetWeekly = jest.fn();
    wrapper = shallow(
      <Weekly currentUser={mockStore} setWeeklyStats={mockSetWeekly} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set state of x and y coordinates', () => {
    const event = { screenX: 23, screenY: 150 };

    wrapper.instance().trackMouseCoordinates(event);

    expect(wrapper.state('xCoordinates')).toEqual(23);
    expect(wrapper.state('yCoordinates')).toEqual(150);
  });

  it('should set proper state on handleClick', () => {
    const initialState = {
      dataDisplay: '',
      toggleDisplay: false,
      xCoordinates: 0,
      yCoordinates: 0
    };
    expect(wrapper.state()).toEqual(initialState);

    const event = { pageX: 30, pageY: 100, y: 2 };

    wrapper.instance().handleClick(event);

    expect(wrapper.state('xCoordinates')).toEqual(30);
    expect(wrapper.state('yCoordinates')).toEqual(100);
    expect(wrapper.state('dataDisplay')).toEqual('2 miles');
    expect(wrapper.state('toggleDisplay')).toEqual(true);
  });

  it('should map properly to store', () => {
    const mapped = mapStateToProps(mockStore);

    expect(mapped.currentUser).toEqual(mockStore.currentUser);
  });

  it('should call dispatch function when a function from map dispatch to props is called', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.setWeeklyStats();

    expect(mockDispatch).toHaveBeenCalled();
  });
});
