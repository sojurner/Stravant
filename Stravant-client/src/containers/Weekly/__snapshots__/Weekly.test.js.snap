// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weekly should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Weekly
    currentUser={
        Object {
            "info": Object {
              "accessToken": "a7eeea216a46a760a74c60acbffc3b55c66537c5",
              "firstName": "Paul",
              "gender": "M",
              "profilePic": "https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg",
              "userId": 34515396,
            },
            "totalStats": Object {},
            "weeklyStats": Object {},
          }
    }
    setWeeklyStats={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <h3
          className="weekly-title"
>
          Weekly Progress
</h3>,
        <AreaChart
          areaColors={
                    Array [
                              "orange",
                            ]
          }
          axes={true}
          axisLabels={
                    Object {
                              "x": "",
                              "y": "",
                            }
          }
          clickHandler={[Function]}
          data={
                    Array [
                              Array [],
                            ]
          }
          dataPoints={true}
          datePattern="%d-%b-%y"
          grid={true}
          height={450}
          interpolate="cardinal"
          margin={
                    Object {
                              "bottom": 60,
                              "left": 60,
                              "right": 10,
                              "top": 10,
                            }
          }
          mouseMoveHandler={[Function]}
          mouseOutHandler={[Function]}
          mouseOverHandler={[Function]}
          verticalGrid={true}
          width={375}
          xType="text"
          yDomainRange={
                    Array [
                              0,
                              2.4,
                            ]
          }
          yType="linear"
/>,
        <p
          className="x-axis"
>
           Day of Week 
</p>,
        <p
          className="y-axis"
>
          Distance
</p>,
        <p
          className="area-data-details"
          style={
                    Object {
                              "left": 10,
                              "position": "absolute",
                              "top": -140,
                            }
          }
/>,
      ],
      "className": "compare-data-text",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Weekly Progress",
          "className": "weekly-title",
        },
        "ref": null,
        "rendered": "Weekly Progress",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "areaColors": Array [
            "orange",
          ],
          "axes": true,
          "axisLabels": Object {
            "x": "",
            "y": "",
          },
          "clickHandler": [Function],
          "data": Array [
            Array [],
          ],
          "dataPoints": true,
          "datePattern": "%d-%b-%y",
          "grid": true,
          "height": 450,
          "interpolate": "cardinal",
          "margin": Object {
            "bottom": 60,
            "left": 60,
            "right": 10,
            "top": 10,
          },
          "mouseMoveHandler": [Function],
          "mouseOutHandler": [Function],
          "mouseOverHandler": [Function],
          "verticalGrid": true,
          "width": 375,
          "xType": "text",
          "yDomainRange": Array [
            0,
            2.4,
          ],
          "yType": "linear",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Day of Week ",
          "className": "x-axis",
        },
        "ref": null,
        "rendered": " Day of Week ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Distance",
          "className": "y-axis",
        },
        "ref": null,
        "rendered": "Distance",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "area-data-details",
          "style": Object {
            "left": 10,
            "position": "absolute",
            "top": -140,
          },
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <h3
            className="weekly-title"
>
            Weekly Progress
</h3>,
          <AreaChart
            areaColors={
                        Array [
                                    "orange",
                                  ]
            }
            axes={true}
            axisLabels={
                        Object {
                                    "x": "",
                                    "y": "",
                                  }
            }
            clickHandler={[Function]}
            data={
                        Array [
                                    Array [],
                                  ]
            }
            dataPoints={true}
            datePattern="%d-%b-%y"
            grid={true}
            height={450}
            interpolate="cardinal"
            margin={
                        Object {
                                    "bottom": 60,
                                    "left": 60,
                                    "right": 10,
                                    "top": 10,
                                  }
            }
            mouseMoveHandler={[Function]}
            mouseOutHandler={[Function]}
            mouseOverHandler={[Function]}
            verticalGrid={true}
            width={375}
            xType="text"
            yDomainRange={
                        Array [
                                    0,
                                    2.4,
                                  ]
            }
            yType="linear"
/>,
          <p
            className="x-axis"
>
             Day of Week 
</p>,
          <p
            className="y-axis"
>
            Distance
</p>,
          <p
            className="area-data-details"
            style={
                        Object {
                                    "left": 10,
                                    "position": "absolute",
                                    "top": -140,
                                  }
            }
/>,
        ],
        "className": "compare-data-text",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Weekly Progress",
            "className": "weekly-title",
          },
          "ref": null,
          "rendered": "Weekly Progress",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "areaColors": Array [
              "orange",
            ],
            "axes": true,
            "axisLabels": Object {
              "x": "",
              "y": "",
            },
            "clickHandler": [Function],
            "data": Array [
              Array [],
            ],
            "dataPoints": true,
            "datePattern": "%d-%b-%y",
            "grid": true,
            "height": 450,
            "interpolate": "cardinal",
            "margin": Object {
              "bottom": 60,
              "left": 60,
              "right": 10,
              "top": 10,
            },
            "mouseMoveHandler": [Function],
            "mouseOutHandler": [Function],
            "mouseOverHandler": [Function],
            "verticalGrid": true,
            "width": 375,
            "xType": "text",
            "yDomainRange": Array [
              0,
              2.4,
            ],
            "yType": "linear",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Day of Week ",
            "className": "x-axis",
          },
          "ref": null,
          "rendered": " Day of Week ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Distance",
            "className": "y-axis",
          },
          "ref": null,
          "rendered": "Distance",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "area-data-details",
            "style": Object {
              "left": 10,
              "position": "absolute",
              "top": -140,
            },
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Weekly should match snapshot when there are weekly stats 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Weekly
    currentUser={
        Object {
            "info": Object {
              "accessToken": "a7eeea216a46a760a74c60acbffc3b55c66537c5",
              "firstName": "Paul",
              "gender": "M",
              "profilePic": "https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg",
              "userId": 34515396,
            },
            "totalStats": Object {},
            "weeklyStats": Object {
              "Friday": 3784.3,
              "Monday": 0,
              "Saturday": 0,
              "Sunday": 0,
              "Thursday": 0,
              "Tuesday": 0,
              "Wednesday": 0,
            },
          }
    }
    setWeeklyStats={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <h3
          className="weekly-title"
>
          Weekly Progress
</h3>,
        <AreaChart
          areaColors={
                    Array [
                              "orange",
                            ]
          }
          axes={true}
          axisLabels={
                    Object {
                              "x": "",
                              "y": "",
                            }
          }
          clickHandler={[Function]}
          data={
                    Array [
                              Array [
                                Object {
                                  "x": "Wed",
                                  "y": 0,
                                },
                                Object {
                                  "x": "Tue",
                                  "y": 0,
                                },
                                Object {
                                  "x": "Thu",
                                  "y": 0,
                                },
                                Object {
                                  "x": "Sun",
                                  "y": 0,
                                },
                                Object {
                                  "x": "Sat",
                                  "y": 0,
                                },
                                Object {
                                  "x": "Mon",
                                  "y": 0,
                                },
                                Object {
                                  "x": "Fri",
                                  "y": 2.35,
                                },
                              ],
                            ]
          }
          dataPoints={true}
          datePattern="%d-%b-%y"
          grid={true}
          height={450}
          interpolate="cardinal"
          margin={
                    Object {
                              "bottom": 60,
                              "left": 60,
                              "right": 10,
                              "top": 10,
                            }
          }
          mouseMoveHandler={[Function]}
          mouseOutHandler={[Function]}
          mouseOverHandler={[Function]}
          verticalGrid={true}
          width={375}
          xType="text"
          yDomainRange={
                    Array [
                              0,
                              2.4,
                            ]
          }
          yType="linear"
/>,
        <p
          className="x-axis"
>
           Day of Week 
</p>,
        <p
          className="y-axis"
>
          Distance
</p>,
        <p
          className="area-data-details"
          style={
                    Object {
                              "left": 10,
                              "position": "absolute",
                              "top": -140,
                            }
          }
/>,
      ],
      "className": "compare-data-text",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Weekly Progress",
          "className": "weekly-title",
        },
        "ref": null,
        "rendered": "Weekly Progress",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "areaColors": Array [
            "orange",
          ],
          "axes": true,
          "axisLabels": Object {
            "x": "",
            "y": "",
          },
          "clickHandler": [Function],
          "data": Array [
            Array [
              Object {
                "x": "Wed",
                "y": 0,
              },
              Object {
                "x": "Tue",
                "y": 0,
              },
              Object {
                "x": "Thu",
                "y": 0,
              },
              Object {
                "x": "Sun",
                "y": 0,
              },
              Object {
                "x": "Sat",
                "y": 0,
              },
              Object {
                "x": "Mon",
                "y": 0,
              },
              Object {
                "x": "Fri",
                "y": 2.35,
              },
            ],
          ],
          "dataPoints": true,
          "datePattern": "%d-%b-%y",
          "grid": true,
          "height": 450,
          "interpolate": "cardinal",
          "margin": Object {
            "bottom": 60,
            "left": 60,
            "right": 10,
            "top": 10,
          },
          "mouseMoveHandler": [Function],
          "mouseOutHandler": [Function],
          "mouseOverHandler": [Function],
          "verticalGrid": true,
          "width": 375,
          "xType": "text",
          "yDomainRange": Array [
            0,
            2.4,
          ],
          "yType": "linear",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Day of Week ",
          "className": "x-axis",
        },
        "ref": null,
        "rendered": " Day of Week ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Distance",
          "className": "y-axis",
        },
        "ref": null,
        "rendered": "Distance",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "area-data-details",
          "style": Object {
            "left": 10,
            "position": "absolute",
            "top": -140,
          },
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <h3
            className="weekly-title"
>
            Weekly Progress
</h3>,
          <AreaChart
            areaColors={
                        Array [
                                    "orange",
                                  ]
            }
            axes={true}
            axisLabels={
                        Object {
                                    "x": "",
                                    "y": "",
                                  }
            }
            clickHandler={[Function]}
            data={
                        Array [
                                    Array [
                                      Object {
                                        "x": "Wed",
                                        "y": 0,
                                      },
                                      Object {
                                        "x": "Tue",
                                        "y": 0,
                                      },
                                      Object {
                                        "x": "Thu",
                                        "y": 0,
                                      },
                                      Object {
                                        "x": "Sun",
                                        "y": 0,
                                      },
                                      Object {
                                        "x": "Sat",
                                        "y": 0,
                                      },
                                      Object {
                                        "x": "Mon",
                                        "y": 0,
                                      },
                                      Object {
                                        "x": "Fri",
                                        "y": 2.35,
                                      },
                                    ],
                                  ]
            }
            dataPoints={true}
            datePattern="%d-%b-%y"
            grid={true}
            height={450}
            interpolate="cardinal"
            margin={
                        Object {
                                    "bottom": 60,
                                    "left": 60,
                                    "right": 10,
                                    "top": 10,
                                  }
            }
            mouseMoveHandler={[Function]}
            mouseOutHandler={[Function]}
            mouseOverHandler={[Function]}
            verticalGrid={true}
            width={375}
            xType="text"
            yDomainRange={
                        Array [
                                    0,
                                    2.4,
                                  ]
            }
            yType="linear"
/>,
          <p
            className="x-axis"
>
             Day of Week 
</p>,
          <p
            className="y-axis"
>
            Distance
</p>,
          <p
            className="area-data-details"
            style={
                        Object {
                                    "left": 10,
                                    "position": "absolute",
                                    "top": -140,
                                  }
            }
/>,
        ],
        "className": "compare-data-text",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Weekly Progress",
            "className": "weekly-title",
          },
          "ref": null,
          "rendered": "Weekly Progress",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "areaColors": Array [
              "orange",
            ],
            "axes": true,
            "axisLabels": Object {
              "x": "",
              "y": "",
            },
            "clickHandler": [Function],
            "data": Array [
              Array [
                Object {
                  "x": "Wed",
                  "y": 0,
                },
                Object {
                  "x": "Tue",
                  "y": 0,
                },
                Object {
                  "x": "Thu",
                  "y": 0,
                },
                Object {
                  "x": "Sun",
                  "y": 0,
                },
                Object {
                  "x": "Sat",
                  "y": 0,
                },
                Object {
                  "x": "Mon",
                  "y": 0,
                },
                Object {
                  "x": "Fri",
                  "y": 2.35,
                },
              ],
            ],
            "dataPoints": true,
            "datePattern": "%d-%b-%y",
            "grid": true,
            "height": 450,
            "interpolate": "cardinal",
            "margin": Object {
              "bottom": 60,
              "left": 60,
              "right": 10,
              "top": 10,
            },
            "mouseMoveHandler": [Function],
            "mouseOutHandler": [Function],
            "mouseOverHandler": [Function],
            "verticalGrid": true,
            "width": 375,
            "xType": "text",
            "yDomainRange": Array [
              0,
              2.4,
            ],
            "yType": "linear",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Day of Week ",
            "className": "x-axis",
          },
          "ref": null,
          "rendered": " Day of Week ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Distance",
            "className": "y-axis",
          },
          "ref": null,
          "rendered": "Distance",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "area-data-details",
            "style": Object {
              "left": 10,
              "position": "absolute",
              "top": -140,
            },
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
